cache/refs.go

```diff
@@ -397,14 +396,51 @@ func (sr *immutableRef) Mount(ctx context.Context, readonly bool, s session.Grou
 			rerr error
 		)
 		if err := sr.withRemoteSnapshotLabelsStargzMode(ctx, s, func() {
-			m, rerr = sr.mount(ctx, readonly)
+			m, rerr = sr.mount(ctx)
 		}); err != nil {
 			return nil, err
 		}
 		return m, rerr
 	}
 
-	return sr.mount(ctx, readonly)
+	return sr.mount(ctx)
```

Comment:
`readonly`  doesn't seems in this function. Is it safe to ignore this arg? maybe we need comments somewhere.

Reply:
Yeah this is a pre-existing issue where even though `ImmutableRef` and `MutableRef` both implement the `Mountable` interface, only `MutableRef` can be read-write or read-only. `ImmutableRef` still accepts this readonly parameter in order to satisfy the interface but in practice returns read-only mounts no matter what value is provided.

I added a comment on this line clarifying all this.

I think eventually we could fix this by refactoring the interface, but given it's a pre-existing issue and would require significant rework of code outside the cache package, I don't want to include those updates in this PR.

Reply:
@sipsma Sorry for the slow review. SGTM about adding the commet. I'll do another review today.

Reply:
No worries at all, appreciate it!